### PR TITLE
Genians add new integration

### PR DESCRIPTION
**Pull request by genians-rnd**
## Status
- [ ] In Progress
- [O] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
new integration file

## Description
Use the Genians integration to block IP using assign tag and unassign tag.

Genian NAC must be installed for unit testing.
You can download it by referring to "https://www.genians.com/genian-nac/#trial" , or contact us at "integration@genians.com"

## Screenshots
N/A

## Minimum version of Demisto
- [ ] 5.0.0
- [O] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details: N/A
   - [O] No

## Must have
- [O] Tests
- [O] Documentation 

## Demisto Partner?
- [O] The title must be in the following format: **[genians-1724] short description**

### CODE REVIEW

File: Packs/Genians/ReleaseNotes/1_0_0.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Genians
+- Genians add new integration
```
Comment (guyfreund) [inline Packs/Genians/ReleaseNotes/1_0_0.md:4]:
```suggestion
```

File: Packs/Genians/pack_metadata.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+    "name": "Genians",
+    "description": "Use the Genians integration to block IP using assign tag and unassign tag.",
+    "support": "xsoar",
```
Comment (guyfreund) [inline Packs/Genians/pack_metadata.json:4]:
```suggestion
    "support": "community",
```

File: Packs/Genians/pack_metadata.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    "currentVersion": "1.0.0",
+    "author": "Cortex XSOAR",
```
Comment (guyfreund) [inline Packs/Genians/pack_metadata.json:6]:
```suggestion
    "author": "Genians",
```

File: Packs/Genians/pack_metadata.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    "url": "https://www.paloaltonetworks.com/cortex",
```
Comment (guyfreund) [inline Packs/Genians/pack_metadata.json:7]: Change to Genians url
```suggestion
    "url": "https://www.genians.com/",
```

File: Packs/Genians/pack_metadata.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    "email": "",
```
Comment (guyfreund) [inline Packs/Genians/pack_metadata.json:8]: Change to Genians support
```suggestion
    "email": "support@genians.com",
```

**Review by guyfreund — CHANGES_REQUESTED** [review]

File: Packs/Genians/pack_metadata.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{
+    "name": "Genians",
+    "description": "Use the Genians integration to block IP using assign tag and unassign tag.",
+    "support": "community",
+    "currentVersion": "1.0.0",
+    "author": "Genians",
+    "url": "https://www.genians.com/",
```
Comment (guyfreund) [inline Packs/Genians/pack_metadata.json:7]: @genians-rnd is this a valid url to get support from?

File: Packs/Genians/pack_metadata.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    "email": "support@genians.com",
```
Comment (guyfreund) [inline Packs/Genians/pack_metadata.json:8]: @genians-rnd is this a valid email to get support from?

Comment (genians-rnd) [reply Packs/Genians/pack_metadata.json:8]: Email is integration@genians.com.

I clicked wrong by my mistake.

File: Packs/Genians/pack_metadata.json (hunk 6 above)
Comment (genians-rnd) [reply Packs/Genians/pack_metadata.json:7]: The url is https://docs.genians.com/release/en/
I think the docs page would be nice.

Comment (genians-rnd) [reply Packs/Genians/pack_metadata.json:7]: Can I request 'pull requests' again..?

Comment (guyfreund) [reply Packs/Genians/pack_metadata.json:7]: No need for a new pr, just submit the needed change

File: Packs/Genians/pack_metadata.json (hunk 7 above)
Comment (guyfreund) [reply Packs/Genians/pack_metadata.json:8]: So let's change this accordingly

File: Packs/Genians/pack_metadata.json (hunk 6 above)
Comment (genians-rnd) [reply Packs/Genians/pack_metadata.json]: I changed it.

File: Packs/Genians/pack_metadata.json (hunk 7 above)
Comment (genians-rnd) [reply Packs/Genians/pack_metadata.json]: I changed it.

**Review by guyfreund — APPROVED** [review]

**Comment by guyfreund** [comment]
@genians-rnd we should schedule a quick demo for this integration, what time is possible for you?

**Comment by genians-rnd** [comment]
@guyfreund Do you want a phone or video call?

**Comment by genians-rnd** [comment]
@guyfreund Is the demo done by phone or video phone?

**Comment by guyfreund** [comment]
> @guyfreund Is the demo done by phone or video phone?

Zoom

**Comment by genians-rnd** [comment]
@guyfreund 
OK. I'll talk inside.
Can you wait?

**Comment by guyfreund** [comment]
@genians-rnd, are you on DFIR or prefer to talk of it via email?

**Comment by genians-rnd** [comment]
@guyfreund I am using DFIR. Can you talk through DFIR?

**Comment by guyfreund** [comment]
@genians-rnd what is your DIFR username?

**Comment by genians-rnd** [comment]
@guyfreund Username is 'genians-rnd'

File: Packs/Genians/TestPlaybooks/playbook-Genians-Test.yml [hunk 8]
```diff
@@ -0,0 +1,257 @@
+id: Genians Test
+version: 9
```
Comment (guyfreund) [inline Packs/Genians/TestPlaybooks/playbook-Genians-Test.yml:2]:
```suggestion
version: -1
```

**Review by guyfreund — CHANGES_REQUESTED** [review]

File: Packs/Genians/Integrations/Genians/Genians.yml [hunk 9]
```diff
@@ -0,0 +1,76 @@
+commonfields:
+  id: Genians
+  version: -1
+name: Genians
+display: Genians
+category: Network Security
+description: ip block
```
[the remark this answers is not in the document]
Comment (genians-rnd) [reply Packs/Genians/Integrations/Genians/Genians.yml]: I changed it. Please check again.

File: Packs/Genians/Integrations/Genians/Genians.py [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+    try:
+
+        LOG("Command being called is {0}".format(demisto.command()))
+
+        if demisto.command() == "test-module":
+            demisto.results('ok')
+        elif demisto.command() == 'genians-assign-ip-tag':
+            demisto.results(assign_ip_tag_command())
+        elif demisto.command() == 'genians-unassign-ip-tag':
+            demisto.results(unassign_ip_tag_command())
```
[the remark this answers is not in the document]
Comment (genians-rnd) [reply Packs/Genians/Integrations/Genians/Genians.py]: I changed it. Please check again.

**Comment by guyfreund** [comment]
@genians-rnd build is failing due to not updated docker-image tag & unit test failures, please fix them.
Docker image tag should be raised to `3.8.5.10455`.

**Comment by genians-rnd** [comment]
@guyfreund Changed from Genians.yml to "dockerimage: demisto/python3:3.8.5.10455".

File: Packs/Genians/Integrations/Genians/Genians_test.py [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+@pytest.fixture(autouse=True)
+def init_tests(mocker):
+    mocker.patch.object(demisto, 'params', return_value=integration_params)
+
+
+def test_get_ip_nodeid(mocker):
+    '''Genians-Genian-NAC-get-ip-nodeid'''
+    from Genians import get_ip_nodeid
+
+    res = get_ip_nodeid(mock_demisto_args['ip'])
```
Comment (guyfreund) [inline Packs/Genians/Integrations/Genians/Genians_test.py:33]: You should mock the http request to make the unit test pass

File: Packs/Genians/Integrations/Genians/Genians_test.py [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+
+    assert result_1 != None
+    assert result_2 == mock_demisto_args['ip']
+
+
+def test_get_tag_list(mocker):
+    '''Genians-Genian-NAC-get-tag-list'''
+    from Genians import get_tag_list
+
+    res = get_tag_list()
```
Comment (guyfreund) [inline Packs/Genians/Integrations/Genians/Genians_test.py:46]: You should mock the http request to make the unit test pass

**Review by guyfreund — CHANGES_REQUESTED** [review]

File: Packs/Genians/Integrations/Genians/Genians_test.py (hunk 12 above)
Comment (genians-rnd) [reply Packs/Genians/Integrations/Genians/Genians_test.py]: I changed the code. Please check.

File: Packs/Genians/Integrations/Genians/Genians_test.py (hunk 11 above)
Comment (genians-rnd) [reply Packs/Genians/Integrations/Genians/Genians_test.py]: I changed the code. Please check.

**Review by guyfreund — APPROVED** [review]
Looking good

**Pull request merged**